packages/realm/src/Realm.ts

```diff
@@ -857,11 +859,15 @@ export class Realm {
       throw new Error("Cannot create an object from a detached RealmObject instance");
     }
     if (!Object.values(UpdateMode).includes(mode)) {
-      throw new Error("Unsupported 'updateMode'. Only 'never', 'modified' or 'all' is supported.");
+      throw new Error(
+        `Unsupported 'updateMode'. Only '${UpdateMode.Never}', '${UpdateMode.Modified}' or '${UpdateMode.All}' is supported.`,
+      );
     }
     this.internal.verifyOpen();
     const helpers = this.classes.getHelpers(type);
+    this.currentUpdateMode = mode;
     const realmObject = RealmObject.create(this, values, mode, { helpers });
+    this.currentUpdateMode = undefined;
```

Comment:
This will never happen if `RealmObject.create` throws.

Reply:
This has been updated to use `finally`. Also, see the PR comment further below addressing your longer comment 👍